Penteract/Assets/Scripts/TriggerDialogue.cpp

```diff
@@ -0,0 +1,27 @@
+#include "TriggerDialogue.h"
+
+#include "DialogueManager.h"
+#include "GameplaySystems.h"
+#include "GameObject.h"
+
+EXPOSE_MEMBERS(TriggerDialogue) {
+    MEMBER(MemberType::GAME_OBJECT_UID, gameControllerUID),
+    MEMBER(MemberType::INT, dialogueID)
+};
+
+GENERATE_BODY_IMPL(TriggerDialogue);
+
+void TriggerDialogue::Start() {
+    gameController = GameplaySystems::GetGameObject(gameControllerUID);
+    if (gameController) dialogueManagerScript = GET_SCRIPT(gameController, DialogueManager);
+}
+
+void TriggerDialogue::Update() {}
+    
+void TriggerDialogue::OnCollision(GameObject& /*collidedWith*/, float3 /*collisionNormal*/, float3 /*penetrationDistance*/, void* /*particle*/) {
+
+    if (dialogueManagerScript) {
+        dialogueManagerScript->SetActiveDialogue(&dialogueManagerScript->dialoguesArray[dialogueID]);
```

Comment:
You are still not checking if the dialogueID is less than dialoguesArray.size, otherwise, that'll throw an Exception

Reply:
nice observation